### PR TITLE
Introduced groovy and added ExclusiveJoinTest case to verify the work…

### PR DESCRIPTION
**Pull request by pctreddy**
…ings in the build

### CODE REVIEW

File: test-harness/src/test/groovy/com/netflix/counductor/integration/test/KafkaTaskBasedWorkflowSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+        wft.workflowTaskType = TaskType.KAFKA_PUBLISH
+        wft.taskReferenceName = "t0"
+        templateWf.tasks.add(wft)
+        templateWf.schemaVersion = 2
+        metadataService.registerWorkflowDef(templateWf)
+    }
+
+
+
+    /*def value = ['inputPaths'    : ['${workflow.input.path1}', '${workflow.input.path2}'],
```
Comment (kishorebanala) [inline test-harness/src/test/groovy/com/netflix/counductor/integration/test/KafkaTaskBasedWorkflowSpec.groovy:196]: leftovers?

File: test-harness/src/test/java/com/netflix/conductor/tests/utils/WorkflowCleanUpUtil.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.commons.lang.StringUtils;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.FileOutputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Singleton
+public class WorkflowCleanUpUtil {
```
Comment (kishorebanala) [inline test-harness/src/test/java/com/netflix/conductor/tests/utils/WorkflowCleanUpUtil.java:17]: Nit: Trying to understand what this class does. JavaDoc?

File: test-harness/src/test/groovy/com/netflix/counductor/integration/test/KafkaTaskBasedWorkflowSpec.groovy (hunk 1 above)
Comment (pctreddy) [reply test-harness/src/test/groovy/com/netflix/counductor/integration/test/KafkaTaskBasedWorkflowSpec.groovy]: Cleaned up

File: test-harness/src/test/java/com/netflix/conductor/tests/utils/WorkflowCleanUpUtil.java (hunk 2 above)
Comment (pctreddy) [reply test-harness/src/test/java/com/netflix/conductor/tests/utils/WorkflowCleanUpUtil.java]: Moved to another utility class : WorkflowTestUtil and added documentation

**Pull request merged**